x/slashing/migrations/v4/migrate.go

```diff
@@ -0,0 +1,155 @@
+package v4
+
+import (
+	"cosmossdk.io/errors"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/bits-and-blooms/bitset"
+	gogotypes "github.com/cosmos/gogoproto/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/slashing/types"
+)
+
+// Migrate migrates state to consensus version 4. Specifically, the migration
+// deletes all existing validator bitmap entries and replaces them with a real
+// "chunked" bitmap.
+func Migrate(ctx sdk.Context, cdc codec.BinaryCodec, store storetypes.KVStore, params types.Params) error {
+	// Get all the missed blocks for each validator, based on the existing signing
+	// info.
+	missedBlocks := make([]types.ValidatorMissedBlocks, 0)
+	iterateValidatorSigningInfos(ctx, cdc, store, func(addr sdk.ConsAddress, info types.ValidatorSigningInfo) (stop bool) {
+		bechAddr := addr.String()
+		localMissedBlocks := GetValidatorMissedBlocks(ctx, cdc, store, addr, params)
+
+		missedBlocks = append(missedBlocks, types.ValidatorMissedBlocks{
+			Address:      bechAddr,
+			MissedBlocks: localMissedBlocks,
+		})
+
+		return false
+	})
+
+	// For each missed blocks entry, of which there should only be one per validator,
+	// we clear all the old entries and insert the new chunked entry.
+	for _, mb := range missedBlocks {
```

Comment:
Can or should this loop be inlined into the callback to `iterateValidatorSigningInfos` above? `err` can be captured in a closure and respected by stopping iteration early.

Reply:
No, because it's best not to perform writes over an IAVL iterator (reads). I'm like 99% sure this behavior is not supported.